### PR TITLE
Handle wgsl errors in the game of life example

### PR DESCRIPTION
**Pull request by kornelski**
Currently copypasting the example into a new project without also copying "shaders/game_of_life.wgsl" gives an unhelpful blank screen. 
This change makes it panic instead. I think nicer error handling is outside scope of the example, and this is good enough to point out that the shader code is missing.

### CODE REVIEW

File: examples/shader/compute_shader_game_of_life.rs [hunk 1]
```diff
@@ -222,10 +222,12 @@ impl render_graph::Node for GameOfLifeNode {
...
             GameOfLifeState::Loading => {
-                if let CachedPipelineState::Ok(_) =
-                    pipeline_cache.get_compute_pipeline_state(pipeline.init_pipeline)
-                {
-                    self.state = GameOfLifeState::Init;
+                match pipeline_cache.get_compute_pipeline_state(pipeline.init_pipeline) {
+                    CachedPipelineState::Ok(_) => {
+                        self.state = GameOfLifeState::Init;
+                    }
+                    CachedPipelineState::Err(err) => panic!("{err}"),
```
Comment (alice-i-cecile) [inline examples/shader/compute_shader_game_of_life.rs:229]: Can you add a comment here warning users that they should copy-paste the shader too?

**Comment by kornelski** [comment]
This affects other examples, so if this approach is okay, I can repeat it for other examples too.

**Review by IceSentry — APPROVED** [review]

**Comment by IceSentry** [comment]
> This affects other examples

IIRC game_of_life is the only example that does that kind of thing, I'm not sure which other example you are referring to.

**Comment by kornelski** [comment]
I mean there's a pattern of `examples/foo.rs` using `assets/shaders/foo.rs`

https://github.com/bevyengine/bevy/blob/main/examples/shader/array_texture.rs

https://github.com/bevyengine/bevy/blob/09fb6b48209e7a6b02878ac97fe3c42ad1f189ef/examples/shader/array_texture.rs#L92

https://github.com/bevyengine/bevy/blob/main/assets/shaders/array_texture.wgsl

and the example code run outside of bevy's directory structure fails to load the files it depends on, and displays a blank screen:

> ERROR bevy_asset::server: Path not found: /tmp/assets/shaders/array_texture.wgsl

I'm proposing to move these paths to a `const` near the top of the file, so that users will more likely notice the dependency.

**Review by mockersf — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
I'd be happy to see this done more broadly in some form. The current PR is an improvement though so I'll merge it on Monday unless you tell me not to.

**Comment by rparrett** [comment]
I think moving the shader path into a constant is a great idea.

The actual error produced here feels confusing to me, which is unfortunate.

```
thread '<unnamed>' panicked at examples/shader/compute_shader_game_of_life.rs:233:25:
Initializing assets/shaders/game_of_life.wgsl:
Pipeline could not be compiled because the following shader is not loaded yet: AssetId<bevy_render::render_resource::shader::Shader>{ index: 0, generation: 0}
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Encountered a panic in system `bevy_render::renderer::render_system`!
```

Specifically, the `not loaded *yet*` part. I wonder if that could be improved (not in this PR), or if there are scenarios where that language actually makes sense?

**Pull request merged**